bedrock/base/templates/macros.html

```diff
@@ -292,13 +292,15 @@ <h3>{{ _('Your download link was sent.') }}</h3>
 
     DO NOT include utm_source in this dictionary! (It's already a required param.)
 #}
-{% macro fxa_email_form(entrypoint, utm_source, style, class_name='fxa-email-form', utm_params={}, form_title='', intro_text='', button_text='', button_class='button button-blue') -%}
+{% macro fxa_email_form(entrypoint, utm_source, style, class_name='fxa-email-form', campaign='', utm_params={}, form_title='', intro_text='', button_text='', button_class='button button-blue') -%}
   {% set service  = 'sync' %}
   <form action="{{ settings.FXA_ENDPOINT }}" data-mozillaonline-action="{{ settings.FXA_ENDPOINT_MOZILLAONLINE }}" id="fxa-email-form" class="{{ class_name }}">
     <input type="hidden" name="action" value="email" />
     <input type="hidden" name="context" value="fx_desktop_v3" />
     <input type="hidden" name="service" value="{{ service }}" />
     <input type="hidden" name="entrypoint" value="{{ entrypoint }}" id="fxa-email-form-entrypoint" />
+    <input type="hidden" name="utm_campaign" value="{{ campaign }}-fxa-embedded-form" id="fxa-email-form-utm-campaign" />
+    <input type="hidden" name="utm_medium" value="referral" id="fxa-email-form-utm-medium" />
```

Comment:
If I try and override either of these values it creates a duplicate input instead of overriding the value e.g.

```
utm_params={'campaign': 'accounts-trailhead', 'medium': 'foo'})
```

At this point, I'm not sure if the `for` loop for utm_params still makes sense?

```
{% for attr, val in utm_params.items() %}
    <input type="hidden" name="utm_{{ attr }}" value="{{ val }}" id="fxa-email-form-utm-{{ attr }}" />
{% endfor %}
```

Would it be better to make each utm param explicit in the macro, so we can then make sure values are overridden and not duplicated?

Reply:
Agree with making the utm params that we've been discussing explicit.

As far as the `far` loop. It may need to remain? In `scene1.html` I'm seeing `'term': 'existing-users'` in the `utm_params` items which I am not seeing in other places. I'm not sure what is happening there, but it feels important and other pages in the future may need to add a unique param for whatever reason.

Reply:
I believe there are only a finite set of allowed utm params. Couldn’t we add term as a parameter? It only seems to be used in one place, and the for loop can lead to a confusing state.

Reply:
[According to Google](https://support.google.com), there are 5 utm params you can add. Including `utm_term` which is used to "identify paid search keywords".
So it makes total sense to just make each param explicit. Tossing the loop.